test/functional/test-url-replacements.js

```diff
@@ -145,6 +146,78 @@ describe('UrlReplacements', () => {
     });
   });
 
+  it('should replace PAGE_LOAD_TIME', () => {
+    return expand('?sh=PAGE_LOAD_TIME').then(res => {
+      expect(res).to.match(/sh=\d+/);
+    });
+  });
+
+  it('should replace PAGE_LOAD_TIME if available within a delay', () => {
+    let win = {
+      'performance': {
+        'timing': {
+          'navigationStart': 100,
+          'loadEventStart': 0
+        }
+      },
+      'Object': Object
+    };
+
+    // When the delay does not result in any change.
+    urlReplacementsFor(win).expand('?sh=PAGE_LOAD_TIME&s').then(res => {
```

Comment:
And, of course, you can simply return a new promise from `then` to chain promises - the test would, of course, wait for all of them to complete.